tests/selenium/conftest.py

```diff
@@ -0,0 +1,341 @@
+        self.mixin._wait_for_table(self.id)
+
+        return self.mixin.find_element(
+            '#{} button.last-page'.format(self.id)
+        ).click()
+
+    def has_pagination(self):
+        self.mixin._wait_for_table(self.id)
+
+        return len(self.mixin.find_elements('.previous-next-container')) != 0
+
+    def has_next_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        el = self.mixin.find_element(
+            '#{} button.next-page'.format(self.id)
+        )
+
+        return el is not None and el.is_enabled()
+
+    def has_prev_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        el = self.mixin.find_element(
+            '#{} button.previous-page'.format(self.id)
+        )
+
+        return el is not None and el.is_enabled()
+
+    def has_first_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        el = self.mixin.find_element(
+            '#{} button.first-page'.format(self.id)
+        )
+
+        return el is not None and el.is_enabled()
+
+    def has_last_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        el = self.mixin.find_element(
+            '#{} button.last-page'.format(self.id)
+        )
+
+        return el is not None and el.is_enabled()
+
+    def click_current_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        return self.mixin.find_element(
+            '#{} input.current-page'.format(self.id)
+        ).click()
+
+    def get_current_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        return self.mixin.find_element(
+            '#{} input.current-page'.format(self.id)
+        ).get_attribute('placeholder')
+
+
+class DataTableFacade(object):
+    @preconditions(_validate_id, _validate_mixin)
+    def __init__(self, id, mixin):
+        self.id = id
+        self.mixin = mixin
+
+        self.cell = DataTableCellFacade(id, mixin)
+        self.column = DataTableColumnFacade(id, mixin)
+        self.paging = DataTablePagingFacade(id, mixin)
+        self.row = DataTableRowFacade(id, mixin)
+
+    def is_ready(self):
+        return self.mixin._wait_for_table(self.id, _READY)
+
+    def is_loading(self):
+        return self.mixin._wait_for_table(self.id, _LOADING)
+
+class DataTableMixin(object):
+    @preconditions(_validate_id, _validate_state)
+    def _wait_for_table(self, id, state = _ANY):
+        return WebDriverWait(self.driver, _TIMEOUT).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR, '#{} {}'.format(id, state)))
+        )
+
+    @preconditions(_validate_id)
+    def table(self, id):
+        return DataTableContext(
+            DataTableFacade(id, self)
+        )
+
+    def copy(self):
+        ActionChains(self.driver).key_down(Keys.CONTROL).send_keys('c').key_up(
+            Keys.CONTROL
+        ).perform()
+
+    def paste(self):
+        ActionChains(self.driver).key_down(Keys.CONTROL).send_keys('v').key_up(
+            Keys.CONTROL
+        ).perform()
```

Comment:
Can this use your nice hold context too? Like
```
with self.hold(Keys.CONTROL):
    self.driver.send_keys('v')
```

Reply:
https://github.com/plotly/dash-table/pull/696/commits/854276e6f807c967ae411af8e5c86d5fd1e7b2f4